src/Core/Grid/DataProvider/DoctrineGridDataProvider.php

```diff
@@ -66,7 +67,23 @@ public function getData(SearchCriteriaInterface $searchCriteria)
         return new GridData(
             $rows,
             $rowsTotal,
-            $searchQueryBuilder->getSQL()
+            $this->getRawQuery($searchQueryBuilder)
         );
     }
+
+    /**
+     * @param QueryBuilder $queryBuilder
+     * @return string
+     */
+    private function getRawQuery(QueryBuilder $queryBuilder)
+    {
+        $query = $queryBuilder->getSQL();
+        $parameters = $queryBuilder->getParameters();
+
+        foreach ($parameters as $pattern => $value) {
+            $query = str_replace(":$pattern", $value, $query);
```

Comment:
A function like this one is maybe better (need refactoring, first copy & paste from stackoverflow :P):
```php
public function interpolateQuery($query, $params) {
    $keys = array();
    $values = $params;

    # build a regular expression for each parameter
    foreach ($params as $key => $value) {
        if (is_string($key)) {
            $keys[] = '/:'.$key.'/';
        } else {
            $keys[] = '/[?]/';
        }

        if (is_string($value))
            $values[$key] = "'" . $value . "'";

        if (is_array($value))
            $values[$key] = "'" . implode("','", $value) . "'";

        if (is_null($value))
            $values[$key] = 'NULL';
    }

    $query = preg_replace($keys, $values, $query);

    return $query;
}
```

Reply:
I don't want to address every use case from the earth until we meet an error, this is how you basically be late with releases and add some dead code in your code.

Reply:
We offer the capability to everyone to add some code with hook or using a custom Filters, so we must be able to parse default values (boolean, string, array, integer at least), it's not every use case, it's the default use case of your function otherwise, it's not returning a raw query :)
More, there is nothing in the design that prevent null, and `[?]` is not a value, just a replacement key in pcre patterns :P

Reply:
> So show me a real use case when the search can pass an array of data in the search form of logs page Grid and I'll update accordingly, tests comprised.

Remember the label of this pull request: this is not "Make a reusable, tested and complete framework to manage every possible parameter of Doctrine query, even the ones that may not comes from a search form of Grid component" :D

Reply:
Unless we can guarantee that the `:pattern` pattern is the only one that will be fed to this code, or that the code won't break down if not, then it must be adapted.

Reply:
Ì can make a constraint to allow only string values, but I don't want to create a "reusable, tested and complete framework to manage every possible parameter of Doctrine query": is it ok for you? This won't break and this won't be supported (for now).

Reply:
If I understood it right, this code tries to parse prepared statements as MySQL would. The problem happens if another kind of token different from `:pattern` is used, like `?`. It's not magic, it's per spec.

Reply:
I don't contest the spec, I'm saying I need a concrete use case so I can update my code accordingly ^^

Reply:
There is no easy solution to get the request sent to MySQL.
We can stick to @mickaelandrieu solution (after rebase), or updating the feature by separating the display of the prepared request and the parameters ready to be applied.

However, I see in the screenshot you apply strings in the request without quotes, and this will break the SQL execution. :smile: